.github/workflows/main.yml

```
@@ -0,0 +1,40 @@
+name: Build
+on:
+  push:
+  pull_request:
+    branches: [ master develop ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: [ '1.14', '1.13', '1.12', '1.11' ]
+    name: Go ${{ matrix.go }}
+    steps:
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Install golint
+        run: |
+          go get -u golang.org/x/lint/golint
+      
+      - name: Checkout 
+        uses: actions/checkout@v2
+
+      - name: Init
+        shell: bash
+        run: go get -v ./...
+
+      - name: Lint
+        shell: bash
+        run: |
+          diff <(gofmt -d .) <(echo -n)
+          go vet -x ./...
+          golint -set_exit_status ./...
+
+      - name: Test
+        shell: bash
+        run: go test -v -race ./...
```

[Nit] Adding a newline to end of document:

```suggestion
        run: go test -v -race ./... 

```